store/sqlstore/post_store.go

```diff
@@ -837,17 +839,11 @@ func (s *SqlPostStore) GetPostsSince(options model.GetPostsSinceOptions, allowFr
 
 	var posts []*model.Post
 
-	replyCountQuery1 := ""
-	replyCountQuery2 := ""
-	if options.SkipFetchThreads {
-		replyCountQuery1 = `, (SELECT COUNT(Posts.Id) FROM Posts WHERE Posts.RootId = (CASE WHEN p1.RootId = '' THEN p1.Id ELSE p1.RootId END) AND Posts.DeleteAt = 0) as ReplyCount`
-		replyCountQuery2 = `, (SELECT COUNT(Posts.Id) FROM Posts WHERE Posts.RootId = (CASE WHEN cte.RootId = '' THEN cte.Id ELSE cte.RootId END) AND Posts.DeleteAt = 0) as ReplyCount`
-	}
 	var query string
 
 	// union of IDs and then join to get full posts is faster in mysql
 	if s.DriverName() == model.DATABASE_DRIVER_MYSQL {
-		query = `SELECT *` + replyCountQuery1 + ` FROM Posts p1 JOIN (
+		query = `SELECT p1.*, COALESCE(T.ReplyCount, 0) as ReplyCount FROM Posts p1 JOIN (
```

Comment:
Did reply counts get migrated somehow or are we going to miss posts from before threads were introduced?

Reply:
Good point, I think this would need a migration, but would be interesting to know if this is going to have a important performance implication or not, because the implication in the performance is minimal, probably the migration itself is not going to be interesting.

Reply:
Other option is to "fallback" to the old approach or "migrate on the fly" whenever is found a non-existing Thread object for a Thread.

Reply:
Sounds good. I am running a couple of load-tests as we speak. Will report soon.